app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -273,21 +275,56 @@ class ExplorationActivityPresenter @Inject constructor(
     explorationDataController.stopPlayingExploration(isCompletion).toLiveData()
       .observe(
         activity,
-        Observer<AsyncResult<Any?>> {
+        {
           when (it) {
             is AsyncResult.Pending -> oppiaLogger.d("ExplorationActivity", "Stopping exploration")
             is AsyncResult.Failure ->
               oppiaLogger.e("ExplorationActivity", "Failed to stop exploration", it.error)
             is AsyncResult.Success -> {
               oppiaLogger.d("ExplorationActivity", "Successfully stopped exploration")
-              backPressActivitySelector()
-              (activity as ExplorationActivity).finish()
+              if (isCompletion) {
+                shouldShowSurveyDialog(profileId, topicId)
+              }
             }
           }
         }
       )
   }
 
+  private fun shouldShowSurveyDialog(profileId: ProfileId, topicId: String) {
+    surveyGatingController.shouldShowSurvey(profileId, topicId).toLiveData()
+      .observe(
+        activity,
+        {
+          when (it) {
+            is AsyncResult.Pending -> {
+              oppiaLogger.d("ExplorationActivity", "Checking survey gating")
+            }
+            is AsyncResult.Failure -> {
+              oppiaLogger.e(
+                "ExplorationActivity",
+                "Failed to retrieve gating decision",
+                it.error
+              )
+            }
+            is AsyncResult.Success -> {
+              if (it.value) {
+                createSurveyDialog().show()
+              } else {
+              }
+            }
+          }
+        }
+      )
+  }
+
+  private fun createSurveyDialog(): AlertDialog {
+    val builder = AlertDialog.Builder(activity)
+    return builder.setTitle("NPS Survey")
+      .setMessage("Survey Intro message")
+      .create()
+  }
```

Comment:
I assume this is temporary as it'll be filled in properly in a later PR? Reminder to remove this code in that case.

Reply:
Yes, I have removed the code.